compiler/src/iree/compiler/Utils/CustomPatternApplicatorPassBase.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_UTILS_CUSTOMPATTERNAPPLICATORPASSBASE_H_
+#define IREE_COMPILER_UTILS_CUSTOMPATTERNAPPLICATORPASSBASE_H_
+
+#include "iree/compiler/Utils/CustomPatternApplicatorPassBase.h"
+#include "llvm/Support/SourceMgr.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/PDL/IR/PDL.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/FileUtilities.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace detail {
+/// Helper to get a list of sizes from the given RankedTensorType value.
```

Comment:
style nit, vertical whitespace
```suggestion

/// Helper to get a list of sizes from the given RankedTensorType value.
```